fix 增加海关规则的同步、修复线上产品map查询异常、优化template组件

fishSteak opened the pull request:
- fix 增加海关规则的同步，在‘产品海关分类表’中触发同步给金蝶+eccang
- 增加新增海关分类时的重复性校验。
- 修改了多个类的命名，将 ErpCustomCategory 和 ErpCustomRule 相关的类名改为 TmsCustomCategory 和 TmsCustomRule
- 优化了部分代码结构，提高了可读性和可维护性
- 删除了未使用的 ErpCustomCategoryMapper 和 ErpCustomRuleMapper 接口
- 更新了模板相关配置和实现，增加了对模板策略的处理

fishSteak committed:
fix 增加海关规则的同步，在‘产品海关分类表’中触发同步给金蝶+eccang

fishSteak committed:
feat(tms): 海关分类重复性校验

- 在 ErpCustomCategoryMapper 中添加 getCustomRuleByMaterialAndDeclaredType 方法- 在 ErpCustomCategoryServiceImpl 中实现 validateCustomRuleCategoryNotExists 方法
- 添加相关单元测试
- 新增 CUSTOM_RULE_CATEGORY_EXISTS 错误码

fishSteak committed:
refactor(yudao-common):调整日志级别并重构 WebUtils 类

- 将 full url 的日志级别从 info改为 debug
- 优化 WebUtils 类中的代码结构

fishSteak committed:
Merge branch 'somle-test' into somle-main-dev-同步报关品名

# Conflicts:
#	somle-esb/src/test/java/com/somle/esb/job/SyncErpProductJobTestSomle.java
#	yudao-module-erp/yudao-module-erp-biz/src/main/java/cn/iocoder/yudao/module/erp/service/product/ErpProductServiceImpl.java
#	yudao-module-tms/yudao-module-tms-api/src/main/java/cn/iocoder/yudao/module/tms/api/logistic/customrule/TmsCustomRuleApi.java
#	yudao-module-tms/yudao-module-tms-biz/src/main/java/cn/iocoder/yudao/module/tms/api/logistic/TmsCustomRuleApiImpl.java
#	yudao-module-tms/yudao-module-tms-biz/src/main/java/cn/iocoder/yudao/module/tms/controller/admin/logistic/category/TmsCustomCategoryController.java
#	yudao-module-tms/yudao-module-tms-biz/src/main/java/cn/iocoder/yudao/module/tms/convert/logistic/TmsCustomRuleConvert.java
#	yudao-module-tms/yudao-module-tms-biz/src/main/java/cn/iocoder/yudao/module/tms/dal/mysql/logistic/category/ErpCustomCategoryMapper.java
#	yudao-module-tms/yudao-module-tms-biz/src/main/java/cn/iocoder/yudao/module/tms/dal/mysql/logistic/customrule/ErpCustomRuleMapper.java
#	yudao-module-tms/yudao-module-tms-biz/src/main/java/cn/iocoder/yudao/module/tms/service/logistic/category/TmsCustomCategoryService.java
#	yudao-module-tms/yudao-module-tms-biz/src/main/java/cn/iocoder/yudao/module/tms/service/logistic/category/TmsCustomCategoryServiceImpl.java
#	yudao-module-tms/yudao-module-tms-biz/src/main/java/cn/iocoder/yudao/module/tms/service/logistic/category/bo/TmsCustomCategoryBO.java
#	yudao-module-tms/yudao-module-tms-biz/src/main/java/cn/iocoder/yudao/module/tms/service/logistic/category/product/TmsCustomProductServiceImpl.java
#	yudao-module-tms/yudao-module-tms-biz/src/main/java/cn/iocoder/yudao/module/tms/service/logistic/customrule/TmsCustomRuleServiceImpl.java

fishSteak committed:
refactor(tms): 重构 TMS 海关分类和规则相关代码

- 修改了多个类的命名，将 ErpCustomCategory 和 ErpCustomRule 相关的类名改为 TmsCustomCategory 和 TmsCustomRule
- 优化了部分代码结构，提高了可读性和可维护性
- 删除了未使用的 ErpCustomCategoryMapper 和 ErpCustomRuleMapper 接口
- 更新了模板相关配置和实现，增加了对模板策略的处理

fishSteak committed:
fix 修复

c-tao commented on yudao-framework/yudao-spring-boot-starter-mq/src/main/java/cn/iocoder/yudao/framework/mq/redis/config/YudaoRedisMQConsumerAutoConfiguration.java:
```diff
@@ -36,9 +37,12 @@
  * @author 芋道源码
  */
 @Slf4j
+@ConditionalOnProperty(name = "spring.task.scheduling.enabled", havingValue = "true", matchIfMissing = true)
```
这个是做什么

fishSteak replied:
这个是控制定时任务的、原本有两个开启定时任务的注解。yudao里面。@EnableScheduling。
现在根据配置文件里面的属性来关闭定时任务。如果没有属性就默认开启

c-tao replied:
spring.task.scheduling.enabled这个属性我查了一下不存在，还是不要自己添加好，会误导小朋友。

fishSteak replied:
都是这么用的，它本身就没考虑关闭的情况。

fishSteak replied:
而且本地执行定时任务，不怕跟线上的token 抢令牌？

fishSteak replied:
原来的并发为0 那个不咋优雅，而且会把任务数据 写入队列，虽然不执行

c-tao commented on yudao-framework/yudao-spring-boot-starter-job/src/main/java/cn/iocoder/yudao/framework/quartz/config/YudaoQuartzAutoConfiguration.java:
```diff
@@ -5,15 +5,14 @@
  */
 @AutoConfiguration
-@EnableScheduling // 开启 Spring 自带的定时任务
+//@EnableScheduling // 开启 Spring 自带的定时任务
```
这个是做什么

fishSteak replied:
yudao @EnableScheduling  重复了

fishSteak committed:
refactor(erp): 重构产品同步逻辑

- 将海关规则同步逻辑移至 ErpProductHandler 类中
- 优化产品同步流程，支持单个产品更新时同步海关规则
- 移除 ErpProductServiceImpl 中的冗余代码
- 更新项目依赖，移除不必要的 yudao-module-tms-api 依赖

c-tao commented on yudao-module-erp/yudao-module-erp-biz/src/main/java/cn/iocoder/yudao/module/erp/service/product/ErpProductServiceImpl.java:
```diff
@@ -195,19 +190,15 @@ public void updateProduct(ErpProductSaveReqVO updateReqVO) {
 
     private void syncProduct(ErpProductDO productDO, Boolean isUpdate) {
         if (isUpdate) {
```
这里好像可以简化

c-tao replied:
DTO的转换要在convert里做

fishSteak replied:
已修改

c-tao commented on yudao-module-srm/yudao-module-srm-biz/src/main/java/cn/iocoder/yudao/module/srm/dal/dataobject/purchase/payment/term/SrmPaymentTermDO.java:
```diff
@@ -42,5 +42,8 @@ public class SrmPaymentTermDO extends BaseDO {
     private String remark;
-
+    /**
+     * 删除
```
这个BaseDO里不是有吗？而且为什么不是继承TenantBaseDO

fishSteak replied:
这个不要逻辑删除 所以取消掉了

fishSteak committed:
refactor(erp): 重构产品同步逻辑

- 简化了产品同步逻辑，移除了冗余代码
- 优化了更新操作中的海关规则处理流程
- 统一了创建人 ID 的设置
- 提高了代码的可读性和可维护性

fishSteak committed:
refactor(srm): 采购支付条款数据对象引入多租户支持

- 将 SrmPaymentTermDO 类的父类从 BaseDO 修改为 TenantBaseDO
-此修改为采购支付条款功能增加了多租户支持，使其能够更好地适应多租户环境

fishSteak committed:
fix 框架修改
